pkg/internal/client/txn.go

```diff
@@ -706,6 +707,11 @@ func (txn *Txn) send(ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.
 	// If we're not yet writing in this txn, but intend to, insert a
 	// begin transaction request before the first write command.
 	if needBeginTxn {
+		// If we're setting the system-config trigger, anchor the transaction to
```

Comment:
We already have `txn.SetSystemConfigTrigger`. Why do we need a separate call to set the txn key for such transactions?